src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1304,11 +1306,12 @@ func (k *kataAgent) createContainer(ctx context.Context, sandbox *Sandbox, c *Co
 
 	// Block based volumes will require some adjustments in the OCI spec, and creation of
 	// storage objects to pass to the agent.
-	volumeStorages, err := k.handleBlkOCIMounts(c, ociSpec)
+	layerStorages, volumeStorages, err := k.handleBlkOCIMounts(c, ociSpec)
```

Comment:
why split out the return into two slices when both of them are appended to ctrStorages anyway?

Reply:
The layers must come before the rootfs because the rootfs is an overlay of the layers (otherwise mounting the rootfs fails). `handleBlkOCIMounts` returns the layers separately so that we can *prepend* them to `ctrStorages`, which makes them be mounted before the rootfs (which is the first entry in `ctrStorages`).

Reply:
Very subtle - could you document this at line 1314

Reply:
Sure, will do.

Reply:
Done in the latest version.